web/assets/css/transmission-app.scss

```diff
@@ -684,73 +706,93 @@ a {
 
-      background: linear-gradient(
-          to right,
-          var(--color-progressbar-leech) 0,
-          var(--color-progressbar-leech) var(--progress, 30%),
-          transparent var(--progress, 30%)
-        ),
-        no-repeat;
+      &::before {
+        background: var(--color-progressbar-leech);
+        color: var(--color-progressbar-fg-2);
+        text-shadow: var(--progress-bar-shadow-2);
+      }
     }
 
-    &.magnet {
-      background: linear-gradient(
-          to right,
-          var(--color-progressbar-magnet) 0,
-          var(--color-progressbar-magnet) var(--progress, 30%),
-          transparent var(--progress, 30%)
-        ),
-        no-repeat;
+    &.magnet::before {
+      background: var(--color-progressbar-magnet);
+      color: #000;
     }
 
     &.seed {
-      &.paused {
-        background-color: var(--color-progressbar-seed-paused);
+      &.paused::before {
+        background: var(--color-progressbar-seed-paused);
+        color: var(--color-progressbar-fg-1);
+        text-shadow: var(--progress-bar-shadow-1);
       }
 
       &:not(.paused) {
-        background: linear-gradient(
-            to right,
-            var(--color-progressbar-seed-1) 0,
-            var(--color-progressbar-seed-1) var(--progress, 30%),
-            var(--color-progressbar-seed-2) var(--progress, 30%)
-          ),
-          no-repeat;
+        &::before {
+          background: var(--color-progressbar-seed-1);
+          color: var(--color-progressbar-fg-2);
+          text-shadow: var(--progress-bar-shadow-2);
+        }
+
+        &::after {
+          background: var(--color-progressbar-seed-2);
+          color: var(--color-progressbar-fg-3);
+          text-shadow: var(--progress-bar-shadow-3);
+        }
+      }
+
+      &.queued::before {
+        background-color: var(--color-progressbar-seed-1);
       }
 
-      &.queued {
-        background-color: var(--color-progressbar-seed-paused);
+      &::before {
+        background-color: var(--color-progressbar-seed-1);
+        color: var(--color-progressbar-fg-2);
+        text-shadow: var(--progress-bar-shadow-2);
       }
     }
 
-    &.verify {
-      background: linear-gradient(
-          to right,
-          var(--color-progressbar-verify) 0,
-          var(--color-progressbar-verify) var(--progress, 30%),
-          transparent var(--progress, 30%)
-        ),
-        no-repeat;
+    &.verify::before {
+      background: var(--color-progressbar-verify);
+      color: #000;
     }
 
-    &.paused {
-      background: linear-gradient(
-          to right,
-          var(--color-progressbar-paused) 0,
-          var(--color-progressbar-paused) var(--progress, 30%),
-          transparent var(--progress, 30%)
-        ),
-        no-repeat;
+    &.paused::before {
+      background: var(--color-progressbar-paused);
+      color: var(--color-progressbar-fg-1);
+      text-shadow: var(--progress-bar-shadow-1);
+    }
+  }
+
+  .torrent-progress-bar {
```

Comment:
These 2 `.torrent-progress-bar` blocks can be merged.

Reply:
Thank you